### ISSUE TITLE
Installing pods in CI (specifically CircleCI)

### ISSUE THREAD

**Issue by raen79** [react-native-community/cli#576]
Hey guys,

I run into no trouble installing pods locally, but when attempting to install the pods in my CI, using the same XCode, Ruby, and Gemfile version, I run into an error that I see has been reported here before:
`Invalid Podfile file: 784: unexpected token at `

Is there anything you can think of that could cause this when running `pod install`. The only other issue that was opened about it: https://github.com/react-native-community/cli/issues/448 does not provide any solutions to this particular problem.

You can find a full trace here (with `verbose`):
```
/Users/distiller/project/node_modules/.bin/react-native config
  {
    "root": "/Users/distiller/project",
    "reactNativePath": "/Users/distiller/project/node_modules/react-native",
    "dependencies": {
      "appcenter": {
        "root": "/Users/distiller/project/node_modules/appcenter",
        "name": "appcenter",
        "platforms": {
          "ios": null,
          "android": null
        },
        "assets": [],
        "hooks": {
          "postlink": "node node_modules/appcenter/scripts/postlink"
        },
        "params": []
      },
      "@react-native-community/async-storage": {
        "root": "/Users/distiller/project/node_modules/@react-native-community/async-storage",
        "name": "@react-native-community/async-storage",
        "platforms": {
          "ios": {
            "sourceDir": "/Users/distiller/project/node_modules/@react-native-community/async-storage/ios",
            "folder": "/Users/distiller/project/node_modules/@react-native-community/async-storage",
            "pbxprojPath": "/Users/distiller/project/node_modules/@react-native-community/async-storage/ios/RNCAsyncStorage.xcodeproj/project.pbxproj",
            "podfile": null,
            "podspecPath": "/Users/distiller/project/node_modules/@react-native-community/async-storage/RNCAsyncStorage.podspec",
            "projectPath": "/Users/distiller/project/node_modules/@react-native-community/async-storage/ios/RNCAsyncStorage.xcodeproj",
            "projectName": "RNCAsyncStorage.xcodeproj",
            "libraryFolder": "Libraries",
            "sharedLibraries": [],
            "plist": []
          },
          "android": {
            "sourceDir": "/Users/distiller/project/node_modules/@react-native-community/async-storage/android",
            "folder": "/Users/distiller/project/node_modules/@react-native-community/async-storage",
            "packageImportPath": "import com.reactnativecommunity.asyncstorage.AsyncStoragePackage;",
            "packageInstance": "new AsyncStoragePackage()"
          }
        },
        "assets": [],
        "hooks": {},
        "params": []
      },
      "@react-native-community/geolocation": {
        "root": "/Users/distiller/project/node_modules/@react-native-community/geolocation",
        "name": "@react-native-community/geolocation",
        "platforms": {
          "ios": {
            "sourceDir": "/Users/distiller/project/node_modules/@react-native-community/geolocation/ios",
            "folder": "/Users/distiller/project/node_modules/@react-native-community/geolocation",
            "pbxprojPath": "/Users/distiller/project/node_modules/@react-native-community/geolocation/ios/RNCGeolocation.xcodeproj/project.pbxproj",
            "podfile": null,
            "podspecPath": "/Users/distiller/project/node_modules/@react-native-community/geolocation/react-native-geolocation.podspec",
            "projectPath": "/Users/distiller/project/node_modules/@react-native-community/geolocation/ios/RNCGeolocation.xcodeproj",
            "projectName": "RNCGeolocation.xcodeproj",
            "libraryFolder": "Libraries",
            "sharedLibraries": [],
            "plist": []
          },
          "android": {
            "sourceDir": "/Users/distiller/project/node_modules/@react-native-community/geolocation/android",
            "folder": "/Users/distiller/project/node_modules/@react-native-community/geolocation",
            "packageImportPath": "import com.reactnativecommunity.geolocation.GeolocationPackage;",
            "packageInstance": "new GeolocationPackage()"
          }
        },
        "assets": [],
        "hooks": {},
        "params": []
      },
      "react-native-code-push": {
        "root": "/Users/distiller/project/node_modules/react-native-code-push",
        "name": "react-native-code-push",
        "platforms": {
          "ios": {
            "sourceDir": "/Users/distiller/project/node_modules/react-native-code-push/ios",
            "folder": "/Users/distiller/project/node_modules/react-native-code-push",
            "pbxprojPath": "/Users/distiller/project/node_modules/react-native-code-push/ios/CodePush.xcodeproj/project.pbxproj",
            "podfile": null,
            "podspecPath": "/Users/distiller/project/node_modules/react-native-code-push/CodePush.podspec",
            "projectPath": "/Users/distiller/project/node_modules/react-native-code-push/ios/CodePush.xcodeproj",
            "projectName": "CodePush.xcodeproj",
            "libraryFolder": "Libraries",
            "sharedLibraries": [
              "libz.tbd"
            ],
            "plist": []
          },
          "android": {
            "sourceDir": "/Users/distiller/project/node_modules/react-native-code-push/android/app",
            "folder": "/Users/distiller/project/node_modules/react-native-code-push",
            "packageImportPath": "import com.microsoft.codepush.react.CodePush;",
            "packageInstance": "new CodePush(getResources().getString(R.string.reactNativeCodePush_androidDeploymentKey), getApplicationContext(), BuildConfig.DEBUG)"
          }
        },
        "assets": [],
        "hooks": {
          "postlink": "node node_modules/react-native-code-push/scripts/postlink/run"
        },
        "params": []
      },
      "react-native-config": {
        "root": "/Users/distiller/project/node_modules/react-native-config",
        "name": "react-native-config",
        "platforms": {
          "ios": {
            "sourceDir": "/Users/distiller/project/node_modules/react-native-config/ios",
            "folder": "/Users/distiller/project/node_modules/react-native-config",
            "pbxprojPath": "/Users/distiller/project/node_modules/react-native-config/ios/ReactNativeConfig.xcodeproj/project.pbxproj",
            "podfile": null,
            "podspecPath": "/Users/distiller/project/node_modules/react-native-config/react-native-config.podspec",
            "projectPath": "/Users/distiller/project/node_modules/react-native-config/ios/ReactNativeConfig.xcodeproj",
            "projectName": "ReactNativeConfig.xcodeproj",
            "libraryFolder": "Libraries",
            "sharedLibraries": [],
            "plist": []
          },
          "android": {
            "sourceDir": "/Users/distiller/project/node_modules/react-native-config/android",
            "folder": "/Users/distiller/project/node_modules/react-native-config",
            "packageImportPath": "import com.lugg.ReactNativeConfig.ReactNativeConfigPackage;",
            "packageInstance": "new ReactNativeConfigPackage()"
          }
        },
        "assets": [],
        "hooks": {},
        "params": []
      },
      "react-native-device-info": {
        "root": "/Users/distiller/project/node_modules/react-native-device-info",
        "name": "react-native-device-info",
        "platforms": {
          "ios": {
            "sourceDir": "/Users/distiller/project/node_modules/react-native-device-info/ios",
            "folder": "/Users/distiller/project/node_modules/react-native-device-info",
            "pbxprojPath": "/Users/distiller/project/node_modules/react-native-device-info/ios/RNDeviceInfo.xcodeproj/project.pbxproj",
            "podfile": null,
            "podspecPath": "/Users/distiller/project/node_modules/react-native-device-info/RNDeviceInfo.podspec",
            "projectPath": "/Users/distiller/project/node_modules/react-native-device-info/ios/RNDeviceInfo.xcodeproj",
            "projectName": "RNDeviceInfo.xcodeproj",
            "libraryFolder": "Libraries",
            "sharedLibraries": [],
            "plist": []
          },
          "android": {
            "sourceDir": "/Users/distiller/project/node_modules/react-native-device-info/android",
            "folder": "/Users/distiller/project/node_modules/react-native-device-info",
            "packageImportPath": "import com.learnium.RNDeviceInfo.RNDeviceInfo;",
            "packageInstance": "new RNDeviceInfo()"
          }
        },
        "assets": [],
        "hooks": {},
        "params": []
      },
      "react-native-doc-viewer": {
        "root": "/Users/distiller/project/node_modules/react-native-doc-viewer",
        "name": "react-native-doc-viewer",
        "platforms": {
          "ios": {
            "sourceDir": "/Users/distiller/project/node_modules/react-native-doc-viewer/ios",
            "folder": "/Users/distiller/project/node_modules/react-native-doc-viewer",
            "pbxprojPath": "/Users/distiller/project/node_modules/react-native-doc-viewer/ios/RNDocViewer.xcodeproj/project.pbxproj",
            "podfile": null,
            "podspecPath": "/Users/distiller/project/node_modules/react-native-doc-viewer/RNReactNativeDocViewer.podspec",
            "projectPath": "/Users/distiller/project/node_modules/react-native-doc-viewer/ios/RNDocViewer.xcodeproj",
            "projectName": "RNDocViewer.xcodeproj",
            "libraryFolder": "Libraries",
            "sharedLibraries": [],
            "plist": []
          },
          "android": {
            "sourceDir": "/Users/distiller/project/node_modules/react-native-doc-viewer/android",
            "folder": "/Users/distiller/project/node_modules/react-native-doc-viewer",
            "packageImportPath": "import com.philipphecht.RNDocViewerPackage;",

[!] Invalid `Podfile` file: 765: unexpected token at '{

  "root": "/Users/distiller/project",

  "reactNativePath": "/Users/distiller/project/node_modules/react-native",

  "dependencies": {

    "appcenter": {

      "root": "/Users/distiller/project/node_modules/appcenter",

      "name": "appcenter",

      "platforms": {

        "ios": null,

        "android": null

      },

      "assets": [],

      "hooks": {

        "postlink": "node node_modules/appcenter/scripts/postlink"

      },

      "params": []

    },

    "@react-native-community/async-storage": {

      "root": "/Users/distiller/project/node_modules/@react-native-community/async-storage",

      "name": "@react-native-community/async-storage",

      "platforms": {

        "ios": {

          "sourceDir": "/Users/distiller/project/node_modules/@react-native-community/async-storage/ios",

          "folder": "/Users/distiller/project/node_modules/@react-native-community/async-storage",

          "pbxprojPath": "/Users/distiller/project/node_modules/@react-native-community/async-storage/ios/RNCAsyncStorage.xcodeproj/project.pbxproj",

          "podfile": null,

          "podspecPath": "/Users/distiller/project/node_modules/@react-native-community/async-storage/RNCAsyncStorage.podspec",

          "projectPath": "/Users/distiller/project/node_modules/@react-native-community/async-storage/ios/RNCAsyncStorage.xcodeproj",

          "projectName": "RNCAsyncStorage.xcodeproj",

          "libraryFolder": "Libraries",

          "sharedLibraries": [],

          "plist": []

        },

        "android": {

          "sourceDir": "/Users/distiller/project/node_modules/@react-native-community/async-storage/android",

          "folder": "/Users/distiller/project/node_modules/@react-native-community/async-storage",

          "packageImportPath": "import com.reactnativecommunity.asyncstorage.AsyncStoragePackage;",

          "packageInstance": "new AsyncStoragePackage()"

        }

      },

      "assets": [],

      "hooks": {},

      "params": []

    },

    "@react-native-community/geolocation": {

      "root": "/Users/distiller/project/node_modules/@react-native-community/geolocation",

      "name": "@react-native-community/geolocation",

      "platforms": {

        "ios": {

          "sourceDir": "/Users/distiller/project/node_modules/@react-native-community/geolocation/ios",

          "folder": "/Users/distiller/project/node_modules/@react-native-community/geolocation",

          "pbxprojPath": "/Users/distiller/project/node_modules/@react-native-community/geolocation/ios/RNCGeolocation.xcodeproj/project.pbxproj",

          "podfile": null,

          "podspecPath": "/Users/distiller/project/node_modules/@react-native-community/geolocation/react-native-geolocation.podspec",

          "projectPath": "/Users/distiller/project/node_modules/@react-native-community/geolocation/ios/RNCGeolocation.xcodeproj",

          "projectName": "RNCGeolocation.xcodeproj",

          "libraryFolder": "Libraries",

          "sharedLibraries": [],

          "plist": []

        },

        "android": {

          "sourceDir": "/Users/distiller/project/node_modules/@react-native-community/geolocation/android",

          "folder": "/Users/distiller/project/node_modules/@react-native-community/geolocation",

          "packageImportPath": "import com.reactnativecommunity.geolocation.GeolocationPackage;",

          "packageInstance": "new GeolocationPackage()"

        }

      },

      "assets": [],

      "hooks": {},

      "params": []

    },

    "react-native-code-push": {

      "root": "/Users/distiller/project/node_modules/react-native-code-push",

      "name": "react-native-code-push",

      "platforms": {

        "ios": {

          "sourceDir": "/Users/distiller/project/node_modules/react-native-code-push/ios",

          "folder": "/Users/distiller/project/node_modules/react-native-code-push",

          "pbxprojPath": "/Users/distiller/project/node_modules/react-native-code-push/ios/CodePush.xcodeproj/project.pbxproj",

          "podfile": null,

          "podspecPath": "/Users/distiller/project/node_modules/react-native-code-push/CodePush.podspec",

          "projectPath": "/Users/distiller/project/node_modules/react-native-code-push/ios/CodePush.xcodeproj",

          "projectName": "CodePush.xcodeproj",

          "libraryFolder": "Libraries",

          "sharedLibraries": [

            "libz.tbd"

          ],

          "plist": []

        },

        "android": {

          "sourceDir": "/Users/distiller/project/node_modules/react-native-code-push/android/app",

          "folder": "/Users/distiller/project/node_modules/react-native-code-push",

          "packageImportPath": "import com.microsoft.codepush.react.CodePush;",

          "packageInstance": "new CodePush(getResources().getString(R.string.reactNativeCodePush_androidDeploymentKey), getApplicationContext(), BuildConfig.DEBUG)"

        }

      },

      "assets": [],

      "hooks": {

        "postlink": "node node_modules/react-native-code-push/scripts/postlink/run"

      },

      "params": []

    },

    "react-native-config": {

      "root": "/Users/distiller/project/node_modules/react-native-config",

      "name": "react-native-config",

      "platforms": {

        "ios": {

          "sourceDir": "/Users/distiller/project/node_modules/react-native-config/ios",

          "folder": "/Users/distiller/project/node_modules/react-native-config",

          "pbxprojPath": "/Users/distiller/project/node_modules/react-native-config/ios/ReactNativeConfig.xcodeproj/project.pbxproj",

          "podfile": null,

          "podspecPath": "/Users/distiller/project/node_modules/react-native-config/react-native-config.podspec",

          "projectPath": "/Users/distiller/project/node_modules/react-native-config/ios/ReactNativeConfig.xcodeproj",


          "projectName": "ReactNativeConfig.xcodeproj",

          "libraryFolder": "Libraries",

          "sharedLibraries": [],

          "plist": []

        },

        "android": {

          "sourceDir": "/Users/distiller/project/node_modules/react-native-config/android",

          "folder": "/Users/distiller/project/node_modules/react-native-config",

          "packageImportPath": "import com.lugg.ReactNativeConfig.ReactNativeConfigPackage;",

          "packageInstance": "new ReactNativeConfigPackage()"

        }

      },

      "assets": [],

      "hooks": {},

      "params": []

    },

    "react-native-device-info": {

      "root": "/Users/distiller/project/node_modules/react-native-device-info",

      "name": "react-native-device-info",

      "platforms": {

        "ios": {

          "sourceDir": "/Users/distiller/project/node_modules/react-native-device-info/ios",

          "folder": "/Users/distiller/project/node_modules/react-native-device-info",

          "pbxprojPath": "/Users/distiller/project/node_modules/react-native-device-info/ios/RNDeviceInfo.xcodeproj/project.pbxproj",

          "podfile": null,

          "podspecPath": "/Users/distiller/project/node_modules/react-native-device-info/RNDeviceInfo.podspec",

          "projectPath": "/Users/distiller/project/node_modules/react-native-device-info/ios/RNDeviceInfo.xcodeproj",

          "projectName": "RNDeviceInfo.xcodeproj",

          "libraryFolder": "Libraries",

          "sharedLibraries": [],

          "plist": []

        },

        "android": {

          "sourceDir": "/Users/distiller/project/node_modules/react-native-device-info/android",

          "folder": "/Users/distiller/project/node_modules/react-native-device-info",

          "packageImportPath": "import com.learnium.RNDeviceInfo.RNDeviceInfo;",

          "packageInstance": "new RNDeviceInfo()"

        }

      },

      "assets": [],

      "hooks": {},

      "params": []

    },

    "react-native-doc-viewer": {

      "root": "/Users/distiller/project/node_modules/react-native-doc-viewer",

      "name": "react-native-doc-viewer",

      "platforms": {

        "ios": {

          "sourceDir": "/Users/distiller/project/node_modules/react-native-doc-viewer/ios",

          "folder": "/Users/distiller/project/node_modules/react-native-doc-viewer",

          "pbxprojPath": "/Users/distiller/project/node_modules/react-native-doc-viewer/ios/RNDocViewer.xcodeproj/project.pbxproj",

          "podfile": null,

          "podspecPath": "/Users/distiller/project/node_modules/react-native-doc-viewer/RNReactNativeDocViewer.podspec",

          "projectPath": "/Users/distiller/project/node_modules/react-native-doc-viewer/ios/RNDocViewer.xcodeproj",

          "projectName": "RNDocViewer.xcodeproj",

          "libraryFolder": "Libraries",

          "sharedLibraries": [],

          "plist": []

        },

        "android": {

          "sourceDir": "/Users/distiller/project/node_modules/react-native-doc-viewer/android",

          "folder": "/Users/distiller/project/node_modules/react-native-doc-viewer",

          "packageImportPath": "import com.philipphecht.RNDocViewerPackage;",
'.

 #  from /Users/distiller/project/ios/Podfile:36
 #  -------------------------------------------
 #  
 >    use_native_modules!
 #  end
 #  -------------------------------------------

/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-core-1.7.4/lib/cocoapods-core/podfile.rb:301:in `rescue in block in from_ruby'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-core-1.7.4/lib/cocoapods-core/podfile.rb:295:in `block in from_ruby'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-core-1.7.4/lib/cocoapods-core/podfile.rb:50:in `instance_eval'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-core-1.7.4/lib/cocoapods-core/podfile.rb:50:in `initialize'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-core-1.7.4/lib/cocoapods-core/podfile.rb:293:in `new'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-core-1.7.4/lib/cocoapods-core/podfile.rb:293:in `from_ruby'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-core-1.7.4/lib/cocoapods-core/podfile.rb:259:in `from_file'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-1.7.4/lib/cocoapods/config.rb:199:in `podfile'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-1.7.4/lib/cocoapods/command.rb:150:in `verify_podfile_exists!'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-1.7.4/lib/cocoapods/command/install.rb:45:in `run'
/Users/distiller/.gem/ruby/2.5.5/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-1.7.4/lib/cocoapods/command.rb:52:in `run'
/Users/distiller/.gem/ruby/2.5.5/gems/cocoapods-1.7.4/bin/pod:55:in `<top (required)>'
/Users/distiller/.gem/ruby/2.5.5/bin/pod:23:in `load'
/Users/distiller/.gem/ruby/2.5.5/bin/pod:23:in `<top (required)>'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:74:in `load'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:74:in `kernel_load'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:28:in `run'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/cli.rb:463:in `exec'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/cli.rb:27:in `dispatch'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/cli.rb:18:in `start'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/exe/bundle:30:in `block in <top (required)>'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/lib/bundler/friendly_errors.rb:124:in `with_friendly_errors'
/Users/distiller/.gem/ruby/2.5.5/gems/bundler-2.0.1/exe/bundle:22:in `<top (required)>'
/Users/distiller/.gem/ruby/2.5.5/bin/bundle:23:in `load'
/Users/distiller/.gem/ruby/2.5.5/bin/bundle:23:in `<main>'
            "packageInstance": "new RNDocViewerPackage()"
          }
        },
        "assets": [],
        "hooks": {},
        "params": []
```

**Comment by thymikee** [comment]
Have you tried [updating the `@react-native-community/cli`](https://github.com/react-native-community/cli#updating-the-cli) to the latest version?

**Comment by raen79** [comment]
@thymikee Yes, my CLI is at 2.7.1.

**Comment by raen79** [comment]
I see someone else had this issue and posted it on the CircleCI forum: https://discuss.circleci.com/t/react-native-0-60-use-native-modules/31474

**Comment by thymikee** [comment]
I'd suspect the terminal buffer output being clipped, and hence invalid JSON. We had these issues on some Windows terminals, where it's very common to have this buffer small.

**Comment by thymikee** [comment]
@alloy is it possible to stream the output and read line-by-line in Ruby like we do in Gradle script?

https://github.com/react-native-community/cli/blob/57fc8ebfb60928f7a82228de3f4b778cf7946854/packages/platform-android/native_modules.gradle#L176-L183

**Comment by thymikee** [comment]
@raen79 can you try if applying this diff (e.g. through `patch-package`) helps?:

```diff
--- a/packages/cli/src/commands/config/config.js
+++ b/packages/cli/src/commands/config/config.js
@@ -27,6 +27,6 @@ export default {
   name: 'config',
   description: 'Print CLI configuration',
   func: async (argv: string[], ctx: ConfigT) => {
-    console.log(JSON.stringify(filterConfig(ctx), null, 2));
+    console.log(JSON.stringify(filterConfig(ctx)));
   },
 };
```

It will make the output smaller

**Comment by raen79** [comment]
I'll have a look, I am also attempting to stream the output.

**Comment by thymikee** [comment]
Fix published in v2.8.0, thank you!
